app/src/main/java/org/merakilearn/datasource/ApplicationRepo.kt

```diff
@@ -154,6 +149,36 @@ class ApplicationRepo(
         }
     }
 
+    suspend fun fetchSavedFile(): List<Pair<String, String>> {
+        return try {
+            withContext(Dispatchers.IO) {
+                val savedFiles = mutableListOf<Pair<String, String>>()
+                val directory = application.applicationContext?.obbDir
+                directory?.listFiles()?.forEach {
+                    if (it.isFile) {
+                        savedFiles.add(Pair(it.path, it.name.split("_")[0]))
+                    }
+                }
+                savedFiles
+            }
+        } catch (ex: Exception) {
+            ex.printStackTrace()
+            emptyList()
+        }
+    }
+
+    suspend fun deleteFile(first: String): Boolean {
```

Comment:
same goes here, lets move it to specific repo

Reply:
Moved to fileDataSource